cmd/qat_plugin/README.md

```diff
@@ -1,39 +1,215 @@
-
-### Prerequisites
-* Data Plane Development Kit (DPDK) drivers must be loaded and configured. For more information, refer to:
-    * [DPDK Getting Started Guide for Linux](https://doc.dpdk.org/guides/linux_gsg/index.html)
-    * [DPDK Getting Started Guide, Linux Drivers section](http://dpdk.org/doc/guides/linux_gsg/linux_drivers.html)
-* QuickAssist SR-IOV virtual functions must be configured. Verify this by running:
-      ```
-      for i in 0442 0443 37c9 19e3; do lspci -d 8086:$i; done
-      ```
-* Intel QuickAssist Technology software for Linux must be installed and
-  configured. For more information, refer to:
-    * [Intel QuickAssist Technology Software for Linux - Getting Started Guide](https://01.org/sites/default/files/downloads/intelr-quickassist-technology/336212qatswgettingstartedguiderev003.pdf)
-    * [Intel QuickAssist Technology on 01.org](https://01.org/intel-quickassist-technology)
-
-### Get source code:
-```
-$ mkdir -p $GOPATH/src/github.com/intel/
-$ cd $GOPATH/src/github.com/intel/
-$ git clone https://github.com/intel/intel-device-plugins-for-kubernetes.git
+# Intel QuickAssist Technology (QAT) device plugin for Kubernetes
+
+# Table of Contents
+
+* [Introduction](#introduction)
+    * [Modes and Configuration options](#modes-and-configuration-options)
+* [Installation](#installation)
+    * [Pre-built image](#pre-built-image)
+    * [Prerequisites](#prerequisites)
+    * [Getting the source code:](#getting-the-source-code)
+    * [Verify node kubelet config](#verify-node-kubelet-config)
+    * [Deploying as a DaemonSet](#deploying-as-a-daemonset)
+        * [Build the plugin image](#build-the-plugin-image)
+        * [Deploy the DaemonSet](#deploy-the-daemonset)
+        * [Verify QAT device plugin is registered on master:](#verify-qat-device-plugin-is-registered-on-master)
+    * [Deploying by hand](#deploying-by-hand)
+        * [Build QAT device plugin](#build-qat-device-plugin)
+        * [Deploy QAT plugin](#deploy-qat-plugin)
+    * [QAT device plugin Demos](#qat-device-plugin-demos)
+        * [DPDK QAT demo](#dpdk-qat-demo)
+            * [Prerequisites](#prerequisites)
+            * [Build the image](#build-the-image)
+            * [Deploy the pod](#deploy-the-pod)
+            * [Run the test](#run-the-test)
+        * [OpenSSL QAT demo](#openssl-qat-demo)
+* [Checking for hardware](#checking-for-hardware)
+
+# Introduction
+
+This Intel QAT device plugin provides support for Intel QAT devices under Kubernetes.
+Supported Devices include:
+
+- [Intel&reg; Xeon&reg; with Intel&reg; C62X Series Chipset][1]
+- [Intel&reg; Atom&trade; Processor C3000][2]
+- [Intel&reg; Communications Chipset 8925 to 8955 Series][3]
+- [Intel&reg; Communications Chipset 8900 to 8920 Series][4]
+- [Intel&reg; Atom&trade; Processor C2000][5]
+
+> **FIXME** - we need to verify this list:
+> - do se support C2000
+> - do we need to add D15xx devices
```

Comment:
my preference is this list covers the list of devices with the VF drivers in the upstream kernel: https://github.com/torvalds/linux/tree/master/drivers/crypto/qat

Reply:
done. also reworded a bit.